flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/sources/TableSourceUtil.scala

```diff
@@ -238,6 +250,14 @@ object TableSourceUtil {
     }
   }
 
+  private def containsTimeAttribute(tableSchema: TableSchema): Boolean = {
+    tableSchema.getWatermarkSpecs.nonEmpty || tableSchema.getTableColumns.exists(isProctime)
+  }
+
+  private def isProctime(column: TableColumn): Boolean = {
+    toScala(column.getExpr).exists(_.equalsIgnoreCase("proctime()"))
+  }
```

Comment:
Actually `proctime(    )` should also work, i'm thinking if we can use the `Sqlparser` to parse the expr first, if we got one `SqlUnresolvedFunction` with no operands and its name matches `proctime`, the we can go ahead.

Reply:
It is hard to get `Sqlparser` too. Obtain parser also creates a circular dependency. It is hard to break it.
I will remove `" "` for checking.

Reply:
I think this is error-prone. The string compare is not a general solution, I can come up with some corner cases: 
```
`proctime()`  // quoted
proctime // function identifier
```

I think a more general solution is pass a `SqlExprToRexConverterFactory` (which can creates a `SqlExprToRexConverter` to get the correct type of an expression) into the `CatalogSchemaTable`. I know it's verbose to pass it into `CatalogSchemaTable`, but I don't find other reasonable ways.

Reply:
done